### ISSUE TITLE
Unused trait methods should create a compile-time warning.

### ISSUE THREAD

**Issue by cedws** [rust-lang/rust#50927]
Take this example code.

```rust
trait MyThing {
    fn do_stuff(&self) -> bool;
    fn do_other_stuff(&self) -> bool;
}

struct Other {}

impl MyThing for Other {
    fn do_stuff(&self) -> bool {
        true
    }

    fn do_other_stuff(&self) -> bool {
        true
    }
}

fn main() {
    let thing = Other {};

    thing.do_stuff();

    // NOTE: We never call `do_other_stuff`.
}
```

Even though we never call `do_other_stuff`, the compiler does not generate a warning that this trait method is unused. We've defined an implementation for it, but this doesn't mean that the method is *used*.

Apologies if this issue has already been raised.

**Comment by steveklabnik** [comment]
triage: no change

**Comment by SebastianJL** [comment]
Just ran into the same thing. Just curious there any plan to implement this?

**Comment by mu001999** [comment]
#118257 fixed this. We can get the output ([play]()):
```
warning: method `do_other_stuff` is never used
 --> src/main.rs:3:8
  |
1 | trait MyThing {
  |       ------- method in this trait
2 |     fn do_stuff(&self) -> bool;
3 |     fn do_other_stuff(&self) -> bool;
  |        ^^^^^^^^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: `playground` (bin "playground") generated 1 warning
```

This can be closed. @steveklabnik
